src/data/nat/mul_ind.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import number_theory.padics.padic_norm
+
+/-!
+# Multiplicative induction principles for ℕ
+
+This file provides three (closely linked) induction principles for ℕ, commonly used for proofs
+about multiplicative functions, such as the totient function and the Möbius function.
+
+## Main definitions
+
+* `nat.rec_on_prime_pow`: Given `P 0, P 1` and a way to extend `P a` to `P (p ^ k * a)`, you can
+  define `P` for all natural numbers.
+* `nat.rec_on_coprime`: Given `P 0`, `P (p ^ k)` for all prime powers, and a way to extend `P a` and
+  `P b` to `P (a * b)` when `a, b` are coprime, you can define `P` for all natural numbers.
+* `nat.rec_on_coprime'`: Given `P 0`, `P 1`, and `P (p ^ k)` for positive prime powers, and a way to
+  extend `P a` and `P b` to `P (a * b)` when `a, b` are coprime,
+  you can define `P` for all natural numbers.
+* `nat.rec_on_completely_multiplicative`: Given `P 0`, `P 1`, `P p` for all primes, and a proof that
+  you can extend `P a` and `P b` to `P (a * b)`, you can define `P` for all natural numbers.
+
+## Implementation notes
+
+The proofs use `padic_val_nat`; however, we have `padic_val_nat p = nat.log p $ nat.gcd k (p ^ k)`
+for any `p ∣ k`, which requires far less imports - the API isn't there though; however, this is why
+it's in `data` even though we import `number_theory`; it's not a particularly deep theorem.
+
+-/
+
+namespace nat
+
+-- REVIEWERS: should this be here? or in `nat.prime`? or should I just replace this with `haveI`?
+instance fact.min_fac_prime (n : ℕ) : fact (prime (n + 2).min_fac) :=
+⟨min_fac_prime $ succ_succ_ne_one n⟩
+
+-- also not sure where these two should be
```

Comment:
Would the `div` section of `data.nat.basic` work for this one, or is that too early?

Reply:
woo! I found both sides of the implication there so less code ^^